web/src/utils/back-and-forth.ts

```diff
@@ -0,0 +1,99 @@
+import { RecordsResult, TopologyMetricsResult } from '../api/loki';
+import { getFlows, getTopologyMetrics } from '../api/routes';
+import { swapFilters } from '../components/filters/filters-helper';
+import { Filter, Filters } from '../model/filters';
+import { FlowQuery, filtersToString } from '../model/flow-query';
+import { TimeRange, computeStepInterval } from './datetime';
+import { mergeStats, substractMetrics, sumMetrics } from './metrics';
+
+export const getFetchFunctions = (filters: Filters, matchAny: boolean) => {
+  // check back-and-forth
+  if (filters.backAndForth) {
+    const swapped = swap(filters.list, matchAny);
+    if (swapped.length > 0) {
+      return {
+        getFlows: (q: FlowQuery) => {
+          return getFlowsBNF(q, filters.list, swapped, matchAny);
+        },
+        getTopologyMetrics: (q: FlowQuery, range: number | TimeRange) => {
+          return getTopologyMetricsBNF(q, range, filters.list, swapped, matchAny);
+        }
+      };
+    }
+  }
+  return {
+    getFlows: getFlows,
+    getTopologyMetrics: getTopologyMetrics
+  };
+};
+
+const encodedPipe = encodeURIComponent('|');
+const getFlowsBNF = (
+  initialQuery: FlowQuery,
+  orig: Filter[],
+  swapped: Filter[],
+  matchAny: boolean
+): Promise<RecordsResult> => {
+  // Combine original filters and swapped. Note that we leave any potential overlapping flows: they can be deduped with "showDuplicates: false".
+  const newFilters = filtersToString(orig, matchAny) + encodedPipe + filtersToString(swapped, matchAny);
+  return getFlows({ ...initialQuery, filters: newFilters });
+};
+
+const getTopologyMetricsBNF = (
+  initialQuery: FlowQuery,
+  range: number | TimeRange,
+  orig: Filter[],
+  swapped: Filter[],
+  matchAny: boolean
+): Promise<TopologyMetricsResult> => {
+  // When bnf is on, this replaces the usual getTopologyMetrics with a function with same arguments that runs 3 queries and merge their results
+  // in order to get the ORIGINAL + SWAPPED - OVERLAP
+  // OVERLAP being ORIGINAL AND SWAPPED.
+  // E.g: if ORIGINAL is "SrcNs=foo", SWAPPED is "DstNs=foo" and OVERLAP is "SrcNs=foo AND DstNs=foo"
```

Comment:
That may increase the `too many outstanding request` issue when not using filters.
See https://issues.redhat.com/browse/NETOBSERV-1235 comments with recording.

Reply:
yes, that's one more query indeed...
I don't see this error often, do you?
we could also start thinking more seriously about a throttling mechanism in the UI ?

Reply:
I haven't retested using loki-operator but it will since the default value is used for outstanding requests.

We already have a story for graphs loadings improvments: https://issues.redhat.com/browse/NETOBSERV-1185 
Let's start from there since it's the page that runs the most of them.

Reply:
i don't remember, do we have a workaround for the "too many outstanding requests" issue? (a loki config?)
Doing some tests on my side, I don't get the error, but I can imagine people may have it.
I can also take a stab at 1185 but that would likely be another PR

Reply:
actually, how about chaining those promises? https://github.com/netobserv/network-observability-console-plugin/pull/379/commits/bc076f028ef17ceccfc6bec43b75cb53abf2b4a2

Reply:
To reproduce you need to remove all filters and set auto refresh to 15s or spam the refresh button.
Then it will appear by itself after some time.
As far as I know `loki-operator` doesn't expose this config.
https://redhat-internal.slack.com/archives/C02939DP5L5/p1693313480687369?thread_ts=1693259801.293669&cid=C02939DP5L5

Chaining the requests would not solve the issue and will make the page load slower right ?
We need to take advantage of parallelization;  Loki `query-frontend` component is designed for that.

Reply:
> Chaining the requests would not solve the issue and will make the page load slower right ?

Make it load slower => yes; but it should reduce the number of parallel queries, which in my understanding is the cause of "too many outstanding requests" error ?

Reply:
fwiw https://github.com/netobserv/network-observability-operator/pull/409 + https://github.com/netobserv/network-observability-console-plugin/pull/380 should make it less likely to get "too many outstanding requests"